File: comments/src/main/java/de/thm/arsnova/service/comment/config/WebSocketConfig.java
package de.thm.arsnova.service.comment.config;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Configuration;
import org.springframework.messaging.simp.config.MessageBrokerRegistry;
import org.springframework.web.socket.config.annotation.EnableWebSocketMessageBroker;
import org.springframework.web.socket.config.annotation.StompEndpointRegistry;
import org.springframework.web.socket.config.annotation.WebSocketMessageBrokerConfigurer;

@Configuration
@EnableWebSocketMessageBroker
public class WebSocketConfig implements WebSocketMessageBrokerConfigurer {
    @Value("${messaging.prefix}") private String messagingPrefix;
    @Value("${stomp.destination.prefix}") private String[] destinationPrefix;
    @Value("${stomp.relay.host}") private String relayHost;
    @Value("${stomp.relay.port}") private int relayPort;
    @Value("${stomp.relay.user}") private String relayUser;
    @Value("${stomp.relay.password}") private String relayPassword;
    @Value("${stomp.user.registry.broadcast}") private String userRegistryBroadcast;
    @Value("${stomp.user.destination.broadcast}") private String userDestinationBroadcast;

    @Override
    public void configureMessageBroker(MessageBrokerRegistry config) {
        config
                .setApplicationDestinationPrefixes(messagingPrefix)
                .enableStompBrokerRelay(destinationPrefix)
                .setUserRegistryBroadcast(userRegistryBroadcast)
                .setUserDestinationBroadcast(userDestinationBroadcast)
                .setRelayHost(relayHost)
                .setRelayPort(relayPort)
                .setClientLogin(relayUser)
                .setClientPasscode(relayPassword);
    }

    @Override
    public void registerStompEndpoints(StompEndpointRegistry registry) {
        registry.addEndpoint("/ws").setAllowedOrigins("*").withSockJS();
    }

}

File: websocket/src/main/kotlin/de/thm/arsnova/service/wsgateway/config/WebSocketConfig.kt
package de.thm.arsnova.service.wsgateway.config

import de.thm.arsnova.service.wsgateway.adapter.AuthChannelInterceptorAdapter
import org.slf4j.LoggerFactory
import org.springframework.boot.context.properties.EnableConfigurationProperties
import org.springframework.context.annotation.Configuration
import org.springframework.messaging.simp.config.ChannelRegistration
import org.springframework.messaging.simp.config.MessageBrokerRegistry
import org.springframework.web.socket.config.annotation.EnableWebSocketMessageBroker
import org.springframework.web.socket.config.annotation.StompEndpointRegistry
import org.springframework.web.socket.config.annotation.WebSocketMessageBrokerConfigurer

@Configuration
@EnableConfigurationProperties(WebSocketProperties::class)
@EnableWebSocketMessageBroker
class WebSocketConfig(
		private val webSocketProperties: WebSocketProperties,
		private val authChannelInterceptorAdapter: AuthChannelInterceptorAdapter
) : WebSocketMessageBrokerConfigurer {
	private val logger = LoggerFactory.getLogger(javaClass)


	override fun configureMessageBroker(config: MessageBrokerRegistry) {
		config
			.setApplicationDestinationPrefixes(webSocketProperties.messagingPrefix)
			.enableStompBrokerRelay(*webSocketProperties.stomp.destinationPrefix)
			.setUserRegistryBroadcast(webSocketProperties.stomp.userRegistryBroadcast)
			.setUserDestinationBroadcast(webSocketProperties.stomp.userDestinationBroadcast)
			.setRelayHost(webSocketProperties.stomp.relay.host)
			.setRelayPort(webSocketProperties.stomp.relay.port)
			.setClientLogin(webSocketProperties.stomp.relay.user)
			.setClientPasscode(webSocketProperties.stomp.relay.password)
	}

	override fun registerStompEndpoints(registry: StompEndpointRegistry) {
		registry.addEndpoint("/ws").withSockJS()
	}


	override fun configureClientInboundChannel(registration: ChannelRegistration) {
		registration.interceptors(authChannelInterceptorAdapter)
	}
}
